lib/galaxy/tools/parameters/basic.py

```diff
@@ -874,23 +874,27 @@ def get_options(self, trans, other_values):
             return self.static_options
 
     def get_legal_values(self, trans, other_values):
-        if self.options:
-            return {v for _, v, _ in self.options.get_options(trans, other_values)}
-        elif self.dynamic_options:
-            try:
-                call_other_values = self._get_dynamic_options_call_other_values(trans, other_values)
-                return {v for _, v, _ in eval(self.dynamic_options, self.tool.code_namespace, call_other_values)}
-            except Exception as e:
-                log.debug("Determining legal values failed for '%s': %s", self.name, unicodify(e))
-                return set()
-        else:
-            return self.legal_values
+        """
+        determine the set of values of legal options
+        """
+        return {v for _, v, _ in self.get_options(trans, other_values)}
+
+    def get_legal_names(self, trans, other_values):
+        """
+        determine a mapping from names to values for all legal options
+        """
+        return {n: v for n, v, _ in self.get_options(trans, other_values)}
 
     def from_json(self, value, trans, other_values={}, require_legal_value=True):
         try:
             legal_values = self.get_legal_values(trans, other_values)
         except ImplicitConversionRequired:
             return value
+        # if the given value is not found in the set of values of the legal
+        # options we fall back to check if the value is in the set of names of
+        # the legal options. this is done with the fallback_values dict which
+        # allows to determine the corresponding legal values
+        fallback_values = self.get_legal_names(trans, other_values)
```

Comment:
Is this something we should add a configuration parameter for - like ``allow_set_by_name`` or something in the parameter definition in the XML?

Reply:
I guess you want to ensure that this does not brake anything?

I would tend to: no. Because

- it already works for statically defined options. so this change just makes this consistent for dynamic options
- a configuration parameter would make this tricky to test with planemo .. there are quite a few tools for which this change allows to add a test for the data tables
- the aim of my implementation (maybe there is still a bug .. see https://github.com/galaxyproject/galaxy/pull/10880#discussion_r546592151) was that values from the name and value columns can't be mixed. I have a hard time finding an example where this could fail. 
- also this applies only to tool tests .. users can not choose from the name column anyway. 

- changing the xml definition might be an option .. but I still tend to no

Reply:
Very good points - the first is particularly convincing.